lib/src/test/java/com/dynatrace/metric/util/MetadataLineFactoryTest.java

```diff
@@ -0,0 +1,130 @@
+                "%s=%s", // unit key = unit value
+            METRIC_NAME, GAUGE_TYPE, MetadataConstants.Dimensions.UNIT_KEY, unit);
+      }
+    }
+  }
+
+  @Test
+  void whenUnitAndDescriptionAreEmpty_shouldReturnNull() {
+    assertNull(MetadataLineFactory.createMetadataLine(METRIC_NAME, null, null, COUNT_TYPE));
+    assertNull(MetadataLineFactory.createMetadataLine(METRIC_NAME, null, null, GAUGE_TYPE));
+  }
+
+  @Test
+  void whenUnitIsNull_shouldCreateMetadataLineWithJustDescription() {
+    assertEquals(
+        createExpectedLine("my\\ description", null),
+        MetadataLineFactory.createMetadataLine(METRIC_NAME, "my description", null, GAUGE_TYPE));
+  }
+
+  @Test
+  void whenUnitIsEmpty_shouldCreateMetadataLineWithJustDescription() {
+    assertEquals(
+        createExpectedLine("my\\ description", null),
+        MetadataLineFactory.createMetadataLine(METRIC_NAME, "my description", "", GAUGE_TYPE));
+  }
+
+  @Test
+  void whenDescriptionIsNull_shouldCreateMetadataLineWithJustUnit() {
+    assertEquals(
+        createExpectedLine(null, "unit"),
+        MetadataLineFactory.createMetadataLine(METRIC_NAME, null, "unit", GAUGE_TYPE));
+  }
+
+  @Test
+  void whenDescriptionIsEmpty_shouldCreateMetadataLineWithJustUnit() {
+    assertEquals(
+        createExpectedLine(null, "unit"),
+        MetadataLineFactory.createMetadataLine(METRIC_NAME, "", "unit", GAUGE_TYPE));
+  }
+
+  @Test
+  void whenDescriptionNeedsEscaping_shouldEscapeDescriptionValue() {
+    assertEquals(
+        createExpectedLine("my\\ description", null),
+        MetadataLineFactory.createMetadataLine(METRIC_NAME, "my description", null, GAUGE_TYPE));
+  }
+
+  @Test
+  void whenUnitAndDescriptionSet_shouldAddBoth() {
+    assertEquals(
+        createExpectedLine("description", "unit"),
+        MetadataLineFactory.createMetadataLine(METRIC_NAME, "description", "unit", GAUGE_TYPE));
+  }
+
+  @Test
+  void bothSet_invalidUnitIgnored() {
+    assertEquals(
+        createExpectedLine("description", null),
+        MetadataLineFactory.createMetadataLine(
+            METRIC_NAME, "description", "{invalid unit}", GAUGE_TYPE));
+  }
+
+  @Test
+  void bothSet_invalidDescriptionIgnored() {
+    assertEquals(
+        createExpectedLine(null, "unit"),
+        MetadataLineFactory.createMetadataLine(METRIC_NAME, "", "unit", GAUGE_TYPE));
+  }
+
+  @Test
+  void bothSet_bothInvalid_shouldReturnNull() {
+    assertNull(
+        MetadataLineFactory.createMetadataLine(METRIC_NAME, "", "{invalid unit}", GAUGE_TYPE));
+  }
+
+  @Test
+  void onlyInvalidDescriptionSet_shouldReturnNull() {
+    assertNull(MetadataLineFactory.createMetadataLine(METRIC_NAME, "", null, GAUGE_TYPE));
+  }
+
+  @Test
+  void onlyInvalidUnitSet_shouldReturnNull() {
+    assertNull(
+        MetadataLineFactory.createMetadataLine(METRIC_NAME, null, "{invalid unit}", GAUGE_TYPE));
+  }
+
+  @Test
+  void addOtherTypeToMetadataLineGeneration_lineIsCreatedWithOtherType() {
```

Comment:
This is a package-internal API, and we might want to extend it in the future. I thought it might be good to check that that works in case we want to add other metric types in the future. We could also restrict it in the createMetadataLine method, but that would mean more code for a hidden API. Either way is fine with me.